homeassistant/components/ovo_energy/sensor.py

```diff
@@ -32,199 +125,54 @@ async def async_setup_entry(
-            "OVO Last Electricity Reading",
-            "mdi:flash",
-            DEVICE_CLASS_ENERGY,
-            "kWh",
-        )
-
-    @property
-    def native_value(self) -> str:
-        """Return the state of the sensor."""
-        usage: OVODailyUsage = self.coordinator.data
-        if usage is None or not usage.electricity:
-            return None
-        return usage.electricity[-1].consumption
-
-    @property
-    def extra_state_attributes(self) -> object:
-        """Return the attributes of the sensor."""
-        usage: OVODailyUsage = self.coordinator.data
-        if usage is None or not usage.electricity:
-            return None
-        return {
-            "start_time": usage.electricity[-1].interval.start,
-            "end_time": usage.electricity[-1].interval.end,
-        }
-
-
-class OVOEnergyLastGasReading(OVOEnergySensor):
-    """Defines a OVO Energy last reading sensor."""
-
-    def __init__(self, coordinator: DataUpdateCoordinator, client: OVOEnergy) -> None:
-        """Initialize OVO Energy sensor."""
-
-        super().__init__(
-            coordinator,
-            client,
-            f"{DOMAIN}_{client.account_id}_last_gas_reading",
-            "OVO Last Gas Reading",
-            "mdi:gas-cylinder",
-            DEVICE_CLASS_ENERGY,
-            "kWh",
-        )
-
-    @property
-    def native_value(self) -> str:
-        """Return the state of the sensor."""
-        usage: OVODailyUsage = self.coordinator.data
-        if usage is None or not usage.gas:
-            return None
-        return usage.gas[-1].consumption
-
-    @property
-    def extra_state_attributes(self) -> object:
-        """Return the attributes of the sensor."""
-        usage: OVODailyUsage = self.coordinator.data
-        if usage is None or not usage.gas:
-            return None
-        return {
-            "start_time": usage.gas[-1].interval.start,
-            "end_time": usage.gas[-1].interval.end,
-        }
-
-
-class OVOEnergyLastElectricityCost(OVOEnergySensor):
-    """Defines a OVO Energy last cost sensor."""
-
-    def __init__(
-        self, coordinator: DataUpdateCoordinator, client: OVOEnergy, currency: str
     ) -> None:
-        """Initialize OVO Energy sensor."""
-        super().__init__(
-            coordinator,
-            client,
-            f"{DOMAIN}_{client.account_id}_last_electricity_cost",
-            "OVO Last Electricity Cost",
-            "mdi:cash-multiple",
-            DEVICE_CLASS_MONETARY,
-            currency,
-        )
+        """Initialize."""
+        if description.key == KEY_LAST_ELECTRICITY_COST:
+            description.native_unit_of_measurement = coordinator.data.electricity[
+                -1
+            ].cost.currency_unit
+        elif description.key == KEY_LAST_GAS_COST:
+            description.native_unit_of_measurement = coordinator.data.gas[
+                -1
+            ].cost.currency_unit
```

Comment:
I would do it in `async_setup_entry`, in a for loop before you create the entities.